server/src/test/java/io/crate/execution/dml/upsert/ShardUpsertRequestTest.java

```diff
@@ -209,4 +219,81 @@ public void test_streaming_item_without_source_from_older_node() throws Exceptio
         ShardUpsertRequest request2 = new ShardUpsertRequest(in);
         assertThat(request2.items().get(0).seqNo()).isEqualTo(SequenceNumbers.SKIP_ON_REPLICA);
     }
+
+    @Test
+    public void test_ram_estimation_with_stats() {
+        Stats stats = new Stats(10L, 1000L, Map.of());
+        Collection<Reference> allColumns = DOC_TABLE_INFO.allColumns();
```

Comment:
Why not use a simple list here instead?
```suggestion
        Collection<Reference> allColumns = List.of(ID_REF, NAME_REF);
```

Reply:
All columns also include `id`, `_raw`, `_doc_id` which should all go into the estimate.